pkg/metrics/metrics.go

```diff
@@ -162,3 +167,15 @@ func isMemoryUnlimited(v uint64) bool {
 	const maxMemorySize = uint64(1 << 62)
 	return v > maxMemorySize
 }
+
+type PodmanMetricsProvider struct {}
+
+func (p *PodmanMetricsProvider) ReadSystemMetrics(ifname string) api.ResourceMetrics {
+	// TODO
```

Comment:
I guess we can reuse here logic from `ItzoMetricsProvider.ReadSystemMetrics` as both itzo and podman runtimes are expected to run on the similar ec2 instances - so gathering system metrics will probably work the same for those two cases. Gathering UnitMetrics will differ.
How about moving contents of  `ItzoMetricsProvider.ReadSystemMetrics` to standalone function and calling it in both Itzo and Podman `ReadSystemMetrics`?

Reply:
That’s a good idea. On it!